lib/zmq-kernel.js

```diff
@@ -177,14 +182,14 @@ export default class ZMQKernel extends Kernel {
         this.options
       );
       this.kernelProcess = spawn;
-      this.setExecutionState("idle");
+      this.monitor(() => {
+        if (onRestarted) onRestarted(this);
```

Comment:
or if we don't want to depend on `this.executionState === 'restarting'` guard.
We can explicitly use independent flag like `this.restarting = true` at entry point of `restart`, then clear `this.restart = false` in monitor callback.

Either way prevent multiple restart request processing.